### PR TITLE
feat: use types repo

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#1307

Imports the types repo in the sdk so we don't have to copy-paste type definitions and maintain them in two different places.
Also exports the types registry used by the API which allows you to create SCALE (chain) types without connecting to the blockchain.

Currently does not provide a mechanism to choose with which types you want to use the sdk (wasn't possible before, not sure we need it). This means that the sdk would likely only ever work with the latest chain version. 

## How to test:

Run against latest develop chain. Will still fail most likely because integration is not fixed, but this shouldn't fail due to types issues.

## Checklist:

- [x] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/chain-helpers/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   },
   "dependencies": {
     "@kiltprotocol/config": "workspace:*",
+    "@kiltprotocol/type-definitions": "0.1.3",
```
Comment (Dudleyneedham) [inline packages/chain-helpers/package.json:29]:
```suggestion
    "@kiltprotocol/type-definitions": "^0.1.3",
```

Would stop us from needing to creating PRs to update it

File: yarn.lock [hunk 2]
```diff
@@ -818,6 +818,7 @@ __metadata:
   resolution: "@kiltprotocol/chain-helpers@workspace:packages/chain-helpers"
   dependencies:
     "@kiltprotocol/config": "workspace:*"
+    "@kiltprotocol/type-definitions": 0.1.3
```
Comment (Dudleyneedham) [inline yarn.lock:821]:
```suggestion
    "@kiltprotocol/type-definitions": ^0.1.3
```

Comment (Dudleyneedham) [reply yarn.lock:821]: Same as above

**Review by Dudleyneedham — CHANGES_REQUESTED** [review]

File: packages/chain-helpers/package.json (hunk 1 above)
Comment (rflechtner) [reply packages/chain-helpers/package.json:29]: oh that's intended, type definitions break to easily with additional work required on the sdk. We'd have to be really careful with semversioning if we're not pinning

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Comment by rflechtner** [comment]
using types8 (mashnet node runtime version 8) for now; this is expected to work flawlessly with the latest release

File: packages/chain-helpers/src/blockchainApiConnection/TypeRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { types8 as KILT_TYPES } from '@kiltprotocol/type-definitions'
+import { TypeRegistry } from '@polkadot/types'
```
Comment (Dudleyneedham) [inline packages/chain-helpers/src/blockchainApiConnection/TypeRegistry.ts:2]:
```suggestion
import type { TypeRegistry } from '@polkadot/types'
```

**Review by Dudleyneedham — CHANGES_REQUESTED** [review]
One minor change regarding `type`

File: packages/chain-helpers/src/blockchainApiConnection/TypeRegistry.ts (hunk 3 above)
Comment (rflechtner) [reply packages/chain-helpers/src/blockchainApiConnection/TypeRegistry.ts:2]: actually this is imported by value, not by type. You see below that we instantiate TypeRegistry

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Pull request merged**